contrib/win32/win32compat/termio.c

```diff
@@ -84,11 +85,44 @@ ReadThread(_In_ LPVOID lpParameter)
 	debug5("TermRead thread, io:%p", pio);
 	memset(&read_status, 0, sizeof(read_status));
 	if (FILETYPE(pio) == FILE_TYPE_CHAR) {
-		while (nBytesReturned == 0) {
-			nBytesReturned = ReadConsoleForTermEmul(WINHANDLE(pio),
-				pio->read_details.buf, pio->read_details.buf_size);
+		if (in_raw_mode) {
+			while (nBytesReturned == 0) {
+				nBytesReturned = ReadConsoleForTermEmul(WINHANDLE(pio),
+					pio->read_details.buf, pio->read_details.buf_size);
+			}
+			read_status.transferred = nBytesReturned;
+		}  else {
+			if (isFirstTime) {
+				isFirstTime = false;
+
+				DWORD dwAttributes;
+				if (!GetConsoleMode(GetStdHandle(STD_INPUT_HANDLE), &dwAttributes))
+					error("GetConsoleMode on STD_INPUT_HANDLE failed with %d\n", GetLastError());
+				
+				dwAttributes |= (ENABLE_ECHO_INPUT | ENABLE_LINE_INPUT | ENABLE_PROCESSED_INPUT);
+
+				if (!SetConsoleMode(GetStdHandle(STD_INPUT_HANDLE), dwAttributes))
```

Comment:
If console attributes are changed, they need to be reset to their original once done. 

Quick fix would be to save the original attributes and revert after ReadLine returns.